### PR TITLE
go.mod: Retract `v1.*` versions

### LINKED ISSUE #2060: Rertact v1.22.0
## Issue

One day we accidentally pushed v1.22.0 tag. It is removed from GitHub Repository but still available in goproxy

```
$ curl -s https://proxy.golang.org/github.com/nspcc-dev/neofs-node/@v/list | grep v1
v1.22.0
```

As the result, when developer imports packages from latest neofs-node, it gets outdated code from v1.22.0.

```
$ go get -u github.com/nspcc-dev/neofs-node@latest
go: downloading github.com/nspcc-dev/neofs-node v1.22.0
go: added github.com/nspcc-dev/neofs-node v1.22.0
```

## How to solve

We can [retract](https://go.dev/ref/mod#go-mod-file-retract) this version. We did this with neofs-api-go in the past: https://github.com/nspcc-dev/neofs-api-go/commit/18a31bc68e57d44e29a25dc3b1fb90eb048f61f8

There is a difference, though. To retract v1.22.0 tag we need to push new tag with bigger value.

> The new version must be higher than other release or pre-release versions; that is, the @latest [version query](https://go.dev/ref/mod#version-queries) should resolve to the new version before retractions are considered.

However we can retract new tag in the same commit, so **both new tag** and **v1.22.0** tag will not be used in the versioning scheme. 

> As an example, consider a case where the author of module example.com/m publishes version v1.0.0 accidentally. To prevent users from upgrading to v1.0.0, the author can add two retract directives to go.mod, then tag v1.0.1 with the retractions.

I suggest to update `go.mod` and push v1.22.1 tag.

```
retract (
	v1.22.0
	v1.22.1
)
```

I did some tests in https://github.com/alexvanin/go-redacted-lib. Go modules use `v0.4.0` as latest version, even though `v1.22.*` tags are present.

### PR DESCRIPTION
**Pull request by fyrchik**
Close #2060 

Signed-off-by: Evgenii Stratonikov <evgeniy@morphbits.ru>

### CODE REVIEW

**Comment by sami-nspcc** [comment]
I have recieved a notification of a new pull request. I am starting the build of images and binaries for further testing.

**Comment by sami-nspcc** [comment]
I have recieved a notification of a new pull request. I am starting the build of images and binaries for further testing.

**Review by carpawell — APPROVED** [review]

**Comment by sami-nspcc** [comment]
I am running integration tests

**Comment by sami-nspcc** [comment]
I am running integration tests

**Comment by sami-nspcc** [comment]
Test run is finished. Please download the tarball from `st*.t5.fs.neo.org`:  FszwwAx3VznDysQGxBUMRj1zj3zp8VcohorVDnunrviV/5bcFMMzRtrvKxoARg2QzrkZxD9Q7CZT2WJv9t6zDfUUq Untar and use `allure open` to watch the report

**Comment by sami-nspcc** [comment]
Test run is finished. Please download the tarball from `st*.t5.fs.neo.org`:  FszwwAx3VznDysQGxBUMRj1zj3zp8VcohorVDnunrviV/6eQfFhtxzBzCCWZjxGczcW34S7Yey29sMEh1TnQRHd5u Untar and use `allure open` to watch the report

File: go.mod [hunk 1]
```diff
@@ -97,3 +97,8 @@ require (
 	gopkg.in/ini.v1 v1.66.6 // indirect
 	gopkg.in/yaml.v2 v2.4.0 // indirect
 )
+
+retract (
+	v1.22.1
+	v1.22.0
```
Comment (alexvanin) [inline go.mod:103]: Some context might be useful.
```
retract (
	v1.22.1 // Published accidentally.
	v1.22.0 // Contains retraction only.
)
```

**Pull request merged**